### PR TITLE
[macOS] resize simple fullscreen windows on windowDidChangeScreen

### PR DESCRIPTION
**Pull request by dylanscott**
I'm an Alacritty user and ran into #1118 (via alacritty/alacritty#2741). It's tagged with "good first issue" and someone helpfully left some details on the AppKit side of things so I thought I'd take a crack at it. I'm kind of a Rust newb (and even bigger AppKit/ObjC newb) so happy for any feedback if this should be tackled in a different way.

- [x] Tested on all platforms changed
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [ ] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [x] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

File: src/platform_impl/macos/window.rs [hunk 1]
```diff
@@ -1339,11 +1340,14 @@ impl WindowExtMacOS for WinitWindow {
             self.set_style_mask_sync(new_mask);
             shared_state_lock.is_simple_fullscreen = false;
 
-            if let Some(presentation_opts) = shared_state_lock.save_presentation_opts {
+            let save_presentation_opts = shared_state_lock.save_presentation_opts;
+            let frame = shared_state_lock.saved_standard_frame();
+            drop(shared_state_lock);
```
Comment (dylanscott) [inline src/platform_impl/macos/window.rs:1345]: when I first added the `windowDidChangeScreen` listener I ran into an issue where we need to acquire the window shared state mutex to check `is_simple_fullscreen`. But calling `app.setPresentationOptions` (which we also do in the above branch / when toggling simple fullscreen on or off) that results in the `windowDidChangeScreen` listener being called synchronously on the same (main) thread.

I wasn't sure the best way to handle this but there were a few other places in this file where the shared state lock is explicitly dropped like this, presumably for similar reasons.

File: src/platform_impl/macos/window_delegate.rs [hunk 2]
```diff
@@ -409,6 +409,17 @@ declare_class!(
                 self.queue_event(WindowEvent::ThemeChanged(theme));
             }
         }
+
+        #[sel(windowDidChangeScreen:)]
+        fn window_did_change_screen(&self, _: Option<&Object>) {
+            trace_scope!("windowDidChangeScreen:");
+            let shared_state = self.window.lock_shared_state("window_did_change_screen");
+            if shared_state.is_simple_fullscreen {
```
Comment (kchibisov) [inline src/platform_impl/macos/window_delegate.rs:417]: That way you'll drop the lock right away.

```suggestion
            let is_simple_fullscreen = self.window.lock_shared_state("window_did_change_screen").is_simple_fullscreen;
            if is_simple_fullscreen {
```

File: src/platform_impl/macos/window.rs [hunk 3]
```diff
@@ -1314,6 +1314,7 @@ impl WindowExtMacOS for WinitWindow {
 
             // Tell our window's state that we're in fullscreen
             shared_state_lock.is_simple_fullscreen = true;
+            drop(shared_state_lock);
```
Comment (kchibisov) [inline src/platform_impl/macos/window.rs:1318]: Add a comment on why you release the lock here.

File: src/platform_impl/macos/window.rs (hunk 1 above)
Comment (kchibisov) [inline src/platform_impl/macos/window.rs:1345]: Will need a comment explaining what you're trying to do here.

**Review by kchibisov — CHANGES_REQUESTED** [review]
Will also need a `CHANGELOG` entry for that.

File: CHANGELOG.md [hunk 4]
```diff
@@ -13,6 +13,7 @@ And please only add new entries to the top of this list, right below the `# Unre
 - On Wayland, fix rare crash on DPI change
 - Web: Added support for `Window::theme`.
 - On Wayland, fix rounding issues when doing resize.
+- On macOS, resize simple fullscreen windows on windowDidChangeScreen events
```
Comment (kchibisov) [inline CHANGELOG.md:16]: Could you revert arbitrary changes you've done to this file besides this line? Thx.

**Review by kchibisov — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (dylanscott) [reply CHANGELOG.md:16]: sorry didn't notice my editor did that

**Review by kchibisov — APPROVED** [review]

**Comment by kchibisov** [comment]
Thanks.

**Pull request merged**